app/models/financialDetails/WhatYouOweChargesList.scala

```diff
@@ -49,12 +50,30 @@ case class WhatYouOweChargesList(balanceDetails: BalanceDetails, chargesList: Li
       && overdueChargeList.exists(_.lpiWithDunningLock.getOrElse[BigDecimal](0) > 0)) true
     else false
 
-
   def interestOnOverdueCharges: Boolean =
     if (overdueChargeList.exists(_.interestOutstandingAmount.isDefined)
       && overdueChargeList.exists(_.latePaymentInterestAmount.getOrElse[BigDecimal](0) <= 0)) true
     else false
 
+  def getRelevantDueDate: LocalDate = {
+    try {
```

Comment:
we don't have a format agreement in this project what we are going to use, either stick to use Try or we can also use try from time to time; so I guess should be to use this approach